app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -655,10 +655,11 @@ private boolean processReturnFromFormEntry(int resultCode, Intent intent) {
 
             // The form is either ready for processing, or not, depending on how it was saved
             if (complete) {
-                // We're honoring in order submissions, now, so trigger a full
-                // submission cycle
+                // Now that we know this form is completed, we can give it the next available
+                // submission ordering number
+                current.setFormNumberForSubmissionOrdering();
```

Comment:
If this is the only place we use this, this is where i'd replace this with something like

current.setFormNumberForSubmissionOrdering(DbUtil.getLargestMetaDataValue(FormRecordStorage, FormRecord.META_SUBMISSION_ORDERING_NUMBER)

which feels like it might isolate the dependency a bit more